### ISSUE TITLE
`rule-name`: no-negated-condition

### ISSUE THREAD

**Issue by BridgeAR** [sindresorhus/eslint-plugin-unicorn#2663]
### Description

I very much like the no-negated-condition rule in almost all cases. There are however cases where the negation improves readability: when having a very short negation condition and a long positive condition. By checking the long condition first, I have to scroll to understand the code.
It would of course be best to prevent such code in the first place and to refactor that while we also all work in code that we have not created and might not be able to refactor all the time.

Adding an option to allow such situations (e.g., with a fixed number of lines for each condition as threshold) would therefore improve the readability in such code bases. It should only apply if the negated side is below e.g., 10 lines and the other side above 10 lines.

### Examples

```js
// ❌
if (!foo) {
  ... 15 lines of code ...
} else {
  ... 3 lines of code ...
}

// ✅
if (!foo) {
  ... 3 lines of code ...
} else {
  ... 15 lines of code ...
}
```

**Comment by sindresorhus** [comment]
I understand the readability concern, but I don’t think we should add this option.

The rule intentionally has a simple contract: negated conditions with an `else` are harder to read, so the rule reports them and can autofix by inverting the condition and swapping the branches. A line-count threshold would make the rule depend on formatting and layout rather than the condition itself, and it raises arbitrary questions around comments, blank lines, nested blocks, and formatter changes.

For the rare cases where the negated branch really is clearer, use an inline disable comment. In larger legacy blocks, refactoring to an early return or extracting the long branch is usually a better fix than making the lint rule layout-aware.
